timeline/plugins.py

```diff
@@ -131,7 +125,7 @@ def search(self, search_terms, user, search_type=None, keywords=None):
             'all': {
                 'title': 'Timeline Events',
                 'search_types': ['timeline'],
-                'items': items,
+                'items': items[:search_limit],
```

Comment:
Here, I would recommend instead using the limit already in `ProjectEventManager.find()`. In this case, we query for the maximum out, convert it into a list, sort that and only then apply the filter. Django querysets are lazy, so if we apply `[:search_limit]` already to `ProjectEvent.objects.filter()`, it will perform much better.

To ensure the returning of results as expected, the sorting by timestamp needs to be moved from here into `find()` as well. So it becomes something like `ProjectEvent.objects.filter(etc etc).order_by('-timestamp')[:search_limit]`.